pkg/issuer/acme/dns/rfc2136/rfc2136_test.go

```diff
@@ -352,11 +357,15 @@ func serverHandlerReturnSuccess(w dns.ResponseWriter, req *dns.Msg) {
 		}
 	}
 
-	w.WriteMsg(m)
+	if err := w.WriteMsg(m); err != nil {
```

Comment:
Thought: Does `panic()` work here? I know panic has special behaviour for `http.ResponseWriter`s but am unsure if it will cause a crash rather than just terminating the handling gorouting.

Reply:
Good point @jakexks . I have now turned these into methods of an object with access to `testing.T` so that I can `assert.NoError` instead, and at least fail the test, if not immediately.